src/Workspaces/Core/Portable/CodeStyle/NotificationOption.cs

```diff
@@ -14,17 +16,28 @@ public class NotificationOption
     {
         public string Name { get; set; }
 
-        public DiagnosticSeverity Value { get; set; }
+        public ReportDiagnostic Severity
```

Comment:
>ReportDiagnostic Severity [](start = 15, length = 25)

I feel this is a good change in the right direction. Now we are correctly representing a rule configuration entry `(string, ReportDiagnostic)`, similar to the [SpecificDiagnosticOptions](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/CompilationOptions.cs,199) used for compilation options generated from ruleset and `nowarn` or `warnaserror` options. However, the key here is the rule name rather than the ID, which means every analyzer has to explicitly get this option and set the effective severity, bail out early if suppressed, etc. If we can eventually move the key to be the rule ID, we can seamlessly merge these workspace options into compilation options at a common place and avoid analyzers to handle these.